### ISSUE TITLE
Republishing Tasks Permathread

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#6900]
Tag the IRC bot with this thread whenever publication resolutions are being discussed.

Agenda+ and add a comment if you want to publish something.

1. Edit this comment to add things to the list. Say **what** is being published (FPWD, WD, CRD, CRS, Rec, etc)
2. Include a link to the latest resolution to publish and/or Agenda+ request for permission to publish, so we can follow-up.
3. Checkmark means has a WG resolution.
4. Link the transition request, once it has been made.
5. Remove from list once actually published.

See also [Transition Requests](https://github.com/w3c/transitions/issues)

Latest:
- [ ] [Revised REC of CSS UI 3](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-3596683647)
- [ ] [New WD of CSS Sizing 3](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-3905493152)
- [ ] [New WD of CSS Sizing 4](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-3905493152)

Backlog of Uncertain Blockedness:
- [ ] Animation Worklet
- [x] Properties & Values API
- [x] RESOLVED: Accept the [PR](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2FTR%2Fcss-contain-2%2F&doc2=https%3A%2F%2Fdrafts.csswg.org%2Fcss-contain-2%2F#cv-notes)  and republishing WD of containment 2
- [x] [WD of Highlight API](https://lists.w3.org/Archives/Public/www-style/2021Nov/0001.html)

**Comment by fantasai** [comment]
Agenda+ to republish Mediaqueries 3 REC editorial update: https://github.com/w3c/csswg-drafts/commits/main/mediaqueries-3

**Comment by frivoal** [comment]
mediaqueries-3 is not written using bikeshed. Does the tooling to generate .html files from .src.html files still exist?

**Comment by svgeesus** [comment]
It still exists but the result requires _so much hand-editing_ that it is basically easier to work with the generated html to add changes. Same issue for Fonts 3, Color 3.

**Comment by fantasai** [comment]
Agenda+ to republish CSS Text Level 4, now that I've merged in css-text-3 and added percentages to letter-spacing and word-spacing (as [resolved](https://github.com/w3c/csswg-drafts/issues/2165) in 2018 T_T)

**Comment by svgeesus** [comment]
> RESOLVED: Republish css-color-4 and css-color-5
Publications requested 22 Apr, expected 28 Apr (after the publishing moratorium ends)

**Comment by svgeesus** [comment]
[CSS Will Change 1 CRD published Thu 5 May](https://www.w3.org/TR/2022/CRD-css-will-change-1-20220505/)

**Comment by svgeesus** [comment]
Publication for CSS Variables 1 CR Snapshot requested 13 June, expected Thu 16 June.

[Republication issue filed](https://github.com/w3c/css-houdini-drafts/issues/1073) on Properties & Values API.

**Comment by svgeesus** [comment]
[Variables 1 CR Snapshot published 16 June 2022](https://www.w3.org/TR/2022/CR-css-variables-1-20220616/)

**Comment by svgeesus** [comment]
CSS Color 4, publication requested 22 June expected Tue 28 June
CSS Color 5, publication requested 23 June expected Tue 28 June

**Comment by svgeesus** [comment]
Agenda+ to publish CRD of CSS Color 4. [Changes since July 2022 CR Snapshot](https://w3c.github.io/csswg-drafts/css-color-4/#changes-from-20220705)

**Comment by fantasai** [comment]
Agenda+ to republish CRS of CSS Text 3 once we approve the last few edits (see open DoC issues).
- DoC: https://drafts.csswg.org/css-text-3/issues-cr-2020
- Changes: https://drafts.csswg.org/css-text-3/#changes

**Comment by svgeesus** [comment]
[CSS Text 3 CRD published 17 Jan 2023](https://www.w3.org/TR/2023/CRD-css-text-3-20230127/)

**Comment by svgeesus** [comment]
Who is doing the [FPWD transition requests](https://github.com/w3c/transitions/issues) for those two?

**Comment by svgeesus** [comment]
> Who is doing the [FPWD transition requests](https://github.com/w3c/transitions/issues) for those two?

Thanks @fantasai 

 - https://github.com/w3c/transitions/issues/487
 - https://github.com/w3c/transitions/issues/488

**Comment by svgeesus** [comment]
[CSS Box 3 published 16 Feb 2023](https://www.w3.org/TR/2023/PR-css-box-3-20230216/)

**Comment by svgeesus** [comment]
[CSS Animations 2 FPWD](https://www.w3.org/TR/2023/WD-css-animations-2-20230302/) and [CSS Animations 1 WD](https://www.w3.org/TR/2023/WD-css-animations-1-20230302/) published today.

**Comment by svgeesus** [comment]
[CSS Text 4](https://www.w3.org/TR/2023/WD-css-text-4-20230301/) published 1 March

**Comment by fantasai** [comment]
Agenda+ to request review and republication of:

- CSS Text 4, which includes final integration of L3 `white-space` with its L4 longhands (adding the newer `break-spaces` value to the `white-space-collapse` longhand and updating prose throughout). See https://drafts.csswg.org/css-text-4/#white-space-property
- CSS Overflow 3, which added a new section defining key terms and concepts around scrolling, see https://drafts.csswg.org/css-overflow-3/#scrolling

(The other changes in this cycle were all in response to WG resolutions on issues.)

**Comment by svgeesus** [comment]
[CSS Box 3 CR Snapshot published 30 Mar](https://www.w3.org/TR/2023/CR-css-display-3-20230330/)

**Comment by sanketj** [comment]
Agenda+ to republish WD for CSS Custom Highlight API 1

Notable changes since last WD:
- Paint behavior clarifications
- I18N warning (follow up from I18N review)

**Comment by svgeesus** [comment]
CSS Box 3 Rec publication requested 03 Apr, expected 06 Apr.

**Comment by svgeesus** [comment]
[CSS 3 Box REC published 6 Apr 2023](https://www.w3.org/TR/2023/REC-css-box-3-20230406/)

**Comment by frivoal** [comment]
Agenda+ to request review and republication of css-text-3 as CRD
* Spec: https://drafts.csswg.org/css-text-3/
* Changes section: https://drafts.csswg.org/css-text-3/#recent-changes
* There are tests for every normative changes since last publication
* open issues: https://github.com/w3c/csswg-drafts/issues?q=is%3Aissue+is%3Aopen+label%3Acss-text-3

**Comment by svgeesus** [comment]
CSS Color 4 has some remaining issues, [23 open 257 closed](https://github.com/w3c/csswg-drafts/labels/css-color-4) but given the large number of [changes since the Candidate Recommendation Draft of 1 November 2022](https://drafts.csswg.org/css-color/#changes-from-20221101) I would like to publish an updated CRD of CSS Color 4.

@LeaVerou @tabatkins 

(CSS Color 5, which is not yet  at CR, [also has a bunch of changes](https://drafts.csswg.org/css-color-5/#changes) so I would republish that at the same time.)

**Comment by svgeesus** [comment]
FPWD of css-viewport-1

 - https://github.com/w3c/csswg-drafts/issues/9799#issuecomment-1894125373

**Comment by svgeesus** [comment]
- https://github.com/w3c/transitions/issues/589

**Comment by svgeesus** [comment]
[CSS Viewport 1 published 25 Jan 2024](https://www.w3.org/TR/2024/WD-css-viewport-1-20240125/)

**Comment by svgeesus** [comment]
Agenda+ For 

1. an **updated WD of CSS Fonts 4**. There remain a fair number of issues [91 open, 248 closed](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-fonts-4) but there have been [a lot of changes](https://drafts.csswg.org/css-fonts-4/#changes-2021-12-21) since the previous /TR publication of [21 December 2021](https://www.w3.org/TR/2021/WD-css-fonts-4-20211221/)
2. an **updated WD of CSS Fonts 5** which has [39 open, 41 closed](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-fonts-5) and a [much smaller list of changes](https://drafts.csswg.org/css-fonts-5/#changes-20211221) but should be republished at the same time, for consistency.
3. [already requested](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-1858885404) CRD update for CSS Color 4 and updated WD of CSS Color 5

@astearns @atanassov these could all be async decisions, of course

**Comment by svgeesus** [comment]
- [x] [CSS Fonts 4 WD published 01 Feb 2024](https://www.w3.org/TR/2024/WD-css-fonts-4-20240201/)
 - [ ]  CSS Fonts 5 WD, publication requested 01 Feb, expected 06 Feb
 - [ ] CSS Color 5 WD in progress

**Comment by fantasai** [comment]
Agenda+ for republishing an **updated draft of CSS Generated Content Level 3** assuming confirmation of edits in #5478, see [Changes List](https://drafts.csswg.org/css-content-3/#changes).

**Comment by astearns** [comment]
The CSSWG will automatically accept this resolution one week from now if no objections are raised here. Anyone can add an emoji to this comment to express support. If you do not support this resolution, please add a new comment.

Proposed Resolution: Publish CRD update for CSS Color 4

**Comment by astearns** [comment]
@fantasai on an updated WD for CSS Generated Content Level 3, you could use https://wiki.csswg.org/spec/publish as the resolution “if there are only non-controversial editorial changes and/or substantive changes approved by the WG”

**Comment by svgeesus** [comment]
> CSS Color 4 has some remaining issues, [23 open 257 closed](https://github.com/w3c/csswg-drafts/labels/css-color-4) but given the large number of [changes since the Candidate Recommendation Draft of 1 November 2022](https://drafts.csswg.org/css-color/#changes-from-20221101) I would like to publish an updated CRD of CSS Color 4.

@astearns could I get an async resolution for CRD on this one, please?

**Comment by astearns** [comment]
> @astearns could I get an async resolution for CRD on this one, please?

Is this not that?

https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-1928584953
https://lists.w3.org/Archives/Public/www-style/2024Feb/0001.html

**Comment by svgeesus** [comment]
Haha I read @fantasai asking for  WD of CSS Generated Content Level 3 and then your async and thought it was in response to her request, not mine.

(Cleans glasses)

Yes, it is.

**Comment by astearns** [comment]
**RESOLVED**: Publish CRD update for CSS Color 4

**Comment by svgeesus** [comment]
CRD of CSS Color 4 [published 13 Feb 2024](Candidate Recommendation Draft)

**Comment by astearns** [comment]
@rachelandrew says we should publish a CRS of css-multicol level 1. There is one change since the last CR publication that is tested. This could be the last draft before we move it to the next stage of the REC track.

**Comment by fantasai** [comment]
Agenda+ for CSS Box L4 WD, see [issue open for confirmation](https://github.com/w3c/csswg-drafts/issues/5132) and [changes list](https://drafts.csswg.org/css-box-4/#changes); and also CSS Box L3 REC [editorial update](https://drafts.csswg.org/css-box-3/#changes).

**Comment by svgeesus** [comment]
@fantasai lmk if you need help with the box 3 rec update, supposedly [echidna can do it](https://www.w3.org/Guide/transitions/?profile=REC&rec=editorial&echidna=true)

**Comment by myakura** [comment]
Sorry if this is not the right place for ask. Are there plans on publishing new Draft for CSS Nesting?

[The latest WD was published over a year ago](https://www.w3.org/TR/2023/WD-css-nesting-1-20230214/) and does not cover the relaxed syntax update.
I know there are other issues (decl shifting-up stuffs) so it not a right time for an update for editors; however, relaxed syntax is already shipped in major engines. I think it's nice the spec being loosely aligned with current implementations and giving heartbeat.

**Comment by svgeesus** [comment]
> relaxed syntax is already shipped in major engines. I think it's nice the spec being loosely aligned with current implementations and giving heartbeat.

Absolutely.

**Comment by svgeesus** [comment]
> @fantasai lmk if you need help with the box 3 rec update, supposedly [echidna can do it](https://www.w3.org/Guide/transitions/?profile=REC&rec=editorial&echidna=true)

Turns out Echidna cannot. Also, the REC boilerplate was incorrect; updated manually.

Publication requested 4 April, expected 11 April due to the AC publishing moratorium.

**Comment by svgeesus** [comment]
WD of CSS Typed OM [published 21 March 2024](https://www.w3.org/TR/2024/WD-css-typed-om-1-20240321/)

**Comment by svgeesus** [comment]
Box 3 Rec [published 11 Apr 2024](https://www.w3.org/TR/2024/REC-css-box-3-20240411/_

**Comment by svgeesus** [comment]
CR Snapshot of [CSS Multicol 1 published 16 May](https://www.w3.org/TR/2024/CR-css-multicol-1-20240516/)

**Comment by frivoal** [comment]
Update REC of CSS-Contain-1 to fold in the 3 proposed corrections.
All have two or more implementations (and have for a while now). https://drafts.csswg.org/css-contain-1/implementation-report-2022-09
There is an open issue (#7875) that is likely to require further changes, but these are additional to what we have already, not a contradiction, so we can just handle it later / in parallel.

**Comment by svgeesus** [comment]
> Requesting updated REC of CSS-Contain-1 to fold in the 3 proposed corrections.

This looks ready to go, to me. +1 to [recommendation update request](https://www.w3.org/Guide/transitions/?profile=REC&rec=substantive)

**Comment by svgeesus** [comment]
[Update request for CSS Contain 1](https://github.com/w3c/transitions/issues/630)

**Comment by svgeesus** [comment]
[CSS Contain 1 republished 25 June 2024](https://www.w3.org/TR/2024/REC-css-contain-1-20240625/)

**Comment by svgeesus** [comment]
Agenda+ for 

1. CSS Conditional 3 to CR Draft. Previous [13 January 2022 Candidate Recommendation Snapshot](https://www.w3.org/TR/2022/CR-css-conditional-3-20220113/) and [changes since](https://drafts.csswg.org/css-conditional-3/#changes).
2. CSS Nesting 1 updated WD as [proposed earlier](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-2027827194) but not discussed. Previous [14 February 2023 WD](https://www.w3.org/TR/2023/WD-css-nesting-1-20230214/) and [changes since](https://drafts.csswg.org/css-nesting/#changes).

**Comment by fantasai** [comment]
@tabatkins and I suggest FPWD of css-values-5

**Comment by svgeesus** [comment]
@fantasai @tabatkins need any help with the FPWD transition request for CSS Values 5?

**Comment by svgeesus** [comment]
[Values 5 FPWD published 3 Sept 2024](https://www.w3.org/TR/2024/WD-css-values-5-20240903/)

**Comment by fantasai** [comment]
Agenda+ to republish [CSS Pseudo-Elements Level 4](https://drafts.csswg.org/css-pseudo-4/) assuming everyone, especially @schenney-chromium and @delan, are satisfied with the edits.

**Comment by noamr** [comment]
Agenda+ to republish WD for [CSS View Transitions Level 2](https://drafts.csswg.org/css-view-transitions-2/)

**Comment by svgeesus** [comment]
> Agenda+ to republish WD for [CSS View Transitions Level 2](https://drafts.csswg.org/css-view-transitions-2/)

There is no changes section, but [there are changes since 16 May](https://github.com/w3c/csswg-drafts/commits/main/css-view-transitions-2/Overview.bs)

**Comment by fantasai** [comment]
Agenda+ to ask conformation for edits and republish css-values-5 with the following changes:

* Defined `if()`, per WG resolution
   <https://drafts.csswg.org/css-values-5/#if-notation>
* Defined `inherit()`, per WG resolution
   <https://drafts.csswg.org/css-values-5/#inherit-notation>
* Redefined `attr()` to match WG resolution (new syntax, and now it's var()-like).
   <https://drafts.csswg.org/css-values-5/#attr-notation>
* Imported definition of "arbitrary substitution function" from Variables. ([Issue 10679](https://github.com/w3c/csswg-drafts/issues/10679))
   <https://drafts.csswg.org/css-values-5/#arbitrary-substitution> 
* Imported the `<syntax>` production from Props & Values API for use in `attr()`.
   <https://drafts.csswg.org/css-values-5/#css-syntax>
* Defined new `<boolean[]>` syntax multiplier. ([Issue 10457](https://github.com/w3c/csswg-drafts/issues/10457))
   <https://drafts.csswg.org/css-values-5/#boolean>

**Comment by svgeesus** [comment]
> Agenda+ to ask conformation for edits and republish css-values-5 with the following changes:

The [changes section](https://drafts.csswg.org/css-values-5/#changes) looks good, +1 to publication

**Comment by svgeesus** [comment]
Adding back Agenda+ (removed by bot) so we don't lose track of:

> Agenda+ to republish WD for [CSS View Transitions Level 2](https://drafts.csswg.org/css-view-transitions-2/)

> Agenda+ to republish [CSS Pseudo-Elements Level 4](https://drafts.csswg.org/css-pseudo-4/)

> Agenda+ CSS Nesting 1 updated WD as [proposed earlier](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-2027827194) but not discussed. Previous [14 February 2023 WD](https://www.w3.org/TR/2023/WD-css-nesting-1-20230214/) and [changes since](https://drafts.csswg.org/css-nesting/#changes).

**Comment by noamr** [comment]
> > Agenda+ to republish WD for [CSS View Transitions Level 2](https://drafts.csswg.org/css-view-transitions-2/)
> 
> There is no changes section, but [there are changes since 16 May](https://github.com/w3c/csswg-drafts/commits/main/css-view-transitions-2/Overview.bs)

Done, let's resolve on this in the next meeting.

**Comment by tabatkins** [comment]
Requesting FPWD of:

* [Display 4](https://drafts.csswg.org/css-display-4/)
* [Overflow 5](https://drafts.csswg.org/css-overflow-5/)
* [Multicol 2](https://drafts.csswg.org/css-multicol-2/)

Meaningful additions:

* Display 4: `reading-flow`, interpolating/animating `display`
* Overflow 5: all the scroll marker-related stuff
* Multicol 2: the `::column` pseudo

**Comment by svgeesus** [comment]
Noting that the last day to request publication is Tues 17 (to be published Thurs 19 Dec) before the publishing moratorium.

**Comment by svgeesus** [comment]
[FPWD of CSS Overflow 5 published 17 Dec](https://www.w3.org/TR/2024/WD-css-overflow-5-20241217/)

**Comment by nt1m** [comment]
I want to request FPWD for CSS Forms 1: https://drafts.csswg.org/css-forms-1/

**Comment by svgeesus** [comment]
> * RESOLVED: Publish FPWD of CSS Forms

[Published 25 March 2025](https://www.w3.org/TR/2025/WD-css-forms-1-20250325/)

**Comment by kbabbitt** [comment]
I'd like to request FPWD of [CSS Gap Decorations level 1](https://drafts.csswg.org/css-gaps-1).

**Comment by svgeesus** [comment]
Publication of CSS Gaps requested 14 April, expected 17 April

**Comment by svgeesus** [comment]
[CSS Gaps published 17 April](https://www.w3.org/TR/2025/WD-css-gaps-1-20250417/)

**Comment by rachelandrew** [comment]
Agenda+ to republish [CSS Display Level 4](https://drafts.csswg.org/css-display-4/) which includes the resolutions to add the [source-order value](https://github.com/w3c/csswg-drafts/issues/11396#issuecomment-2755105190) and [reading-order property](https://github.com/w3c/csswg-drafts/issues/11208#issuecomment-2625850559).

**Comment by bramus** [comment]
Another request, in light of the Viewport Segments Enumeration API [shipping in Chromium soon](https://chromestatus.com/feature/5170498990243840) :

- FPWD of https://drafts.csswg.org/css-env-1/ _(modulo [the preferred text zoom section](https://drafts.csswg.org/css-env-1/#text-zoom), maybe)_
- Re-publication of https://drafts.csswg.org/css-viewport/

**Comment by SebastianZ** [comment]
For the FPWD of CSS Env 1 see #6729.

Sebastian

**Comment by svgeesus** [comment]
<JoshT> RESOLVED: Publish FPWD of CSS Custom Functions
<JoshT> TabAtkins: I'll get publication started.

@tabatkins not seeing FPWD request on https://github.com/w3c/transitions/issues do you need any help?

**Comment by tabatkins** [comment]
No, I was just being slow about it. Filed now, thanks.

**Comment by svgeesus** [comment]
[Mixins FPWD published 15 May 2025](https://www.w3.org/TR/2025/WD-css-mixins-1-20250515/)

**Comment by noamr** [comment]
Requesting to publish an FPWD of [css-shapes-2](https://drafts.csswg.org/css-shapes-2/)

**Comment by fantasai** [comment]
@noamr Strong +1 to FPWD of `shape()` but I'm not sure the rest of the draft is ready.

**Comment by fantasai** [comment]
Also, while we're at it, can we get an updated draft of css-shapes-1? :)

**Comment by noamr** [comment]
> [@noamr](https://github.com/noamr) Strong +1 to FPWD of `shape()` but I'm not sure the rest of the draft is ready.

Hmm you're right. What do we usually do in this situation? Is there a way to publish a draft of part of a spec? I'd probably want to do the same for `corner-shape` in `css-borders-4` soon and `border-shape` is there and not ready.

**Comment by svgeesus** [comment]
> Hmm you're right. What do we usually do in this situation? Is there a way to publish a draft of part of a spec?

Only by splitting it. Options:

- put the not-ready parts into shapes-3, publish fpwd of shapes-2
- put the ready parts into shapes-1, publish updated wd, let shapes-2 get a bit more work before fpwd

**Comment by fantasai** [comment]
@noamr What @svgeesus said. Wrt css-borders-4, the FPWD is "we are working on this idea, it is roughly scoped out like this, and there's general agreement on pursuing this direction (subject to potentially significant changes as we work through feedback)". It's not "we're done designing and ready to ship" -- that's what CR is for. So `border-shape` is totally fine for FPWD.

The hesitation on `shape-inside`/`shape-padding` is because we don't have agreement on fundamental aspects of how it should work (like with automatic sizes).

**Comment by fantasai** [comment]
This [guide to the spec phases](https://fantasai.inkedblade.net/weblog/2011/inside-csswg/process) might be useful. :)

**Comment by noamr** [comment]
> @noamr What @svgeesus said. Wrt css-borders-4, the FPWD is "we are working on this idea, it is roughly scoped out like this, and there's general agreement on pursuing this direction (subject to potentially significant changes as we work through feedback)". It's not "we're done designing and ready to ship" -- that's what CR is for. So `border-shape` is totally fine for FPWD.
> 
> The hesitation on `shape-inside`/`shape-padding` is because we don't have agreement on fundamental aspects of how it should work (like with automatic sizes).

Gotcha, thanks!
So my proposal atm would be to port shape() to css-shapes-1 and publish that.

**Comment by svgeesus** [comment]
> So my proposal atm would be to port shape() to css-shapes-1 and publish that.

I agree that is the best option.

**Comment by noamr** [comment]
Agenda+ for publishing FPWD of css-borders-4

**Comment by noamr** [comment]
> The CSS Working Group just discussed `Republishing Tasks Permathread`, and agreed to the following:
> 
> * `RESOLVED: Publish css-borders-4 with the edits described`
> 
> The full IRC log of that discussion

This is done... @svgeesus @astearns help with publishing the FPWD plz?

**Comment by svgeesus** [comment]
> This is done... [@svgeesus](https://github.com/svgeesus) [@astearns](https://github.com/astearns) help with publishing the FPWD plz?

Sure, I will set it up.

**Comment by svgeesus** [comment]
CSS Borders 4 Publication requested Weds 16 July, expected Tues 22 July

**Comment by svgeesus** [comment]
[CSS Borders 4 published 2 July 2025](https://www.w3.org/TR/2025/WD-css-borders-4-20250722/)

**Comment by alisonmaher** [comment]
Agenda+ to request to republish a CRD for Color Adjust 1 with changes outlined in https://drafts.csswg.org/css-color-adjust-1/#changes

**Comment by svgeesus** [comment]
Color Adjust publication requested Weds 6 Aug expected Tues 12 August 2025 (it was too late for Thurs 7 Aug).

**Comment by svgeesus** [comment]
[Color Adjust published 12 August](https://www.w3.org/TR/2025/CRD-css-color-adjust-1-20250812/)

**Comment by SebastianZ** [comment]
[CSSOM View republished 16 September 2025](https://www.w3.org/TR/2025/WD-cssom-view-1-20250916/)

**Comment by lilles** [comment]
Agenda+ for publishing FPWD of css-anchor-position-2

**Comment by rachelandrew** [comment]
[Back in May I requested re-publication of CSS Display](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-2847854143), this didn't ever happen, can I request again as Chrome has shipped reading-order and it's not in the published spec

**Comment by ydaniv** [comment]
Also asking for publishing WD of web-animations-2

**Comment by svgeesus** [comment]
Anchor Position 2 FPWD Publication requested 17 Oct, expected 21 October 2025

**Comment by svgeesus** [comment]
[Anchor Positioning 2 published 21 Oct 2025](https://www.w3.org/TR/2025/WD-css-anchor-position-2-20251021/)

**Comment by SebastianZ** [comment]
I've just [updated](https://github.com/w3c/csswg-drafts/commit/3d7141a4c16b50dd9a2ac3cb6bbc33923d75316e) the Changes section of CSS Display 4.
@svgeesus With that done, could you please request a new WD? Or is there something missing?

Sebastian

**Comment by svgeesus** [comment]
Updated WD of Display 4 requested 3 November, expected Thurs 6 November

**Comment by svgeesus** [comment]
[CSS Display 4 published 6 Nov 2025](https://www.w3.org/TR/2025/WD-css-display-4-20251106/)

**Comment by SebastianZ** [comment]
I saw that [CSS Content 3](https://www.w3.org/TR/css-content-3/) wasn't published for several years while there were some substantial changes. I've now added the WPTs to it in #13081. And @fantasai thankfully already updated the Changes section.

So I request to publish a new WD on her behalf.

Sebastian

**Comment by SebastianZ** [comment]
Similar to CSS Content 3, the [CSS Tables 3](https://www.w3.org/TR/css-tables-3/) spec wasn't published for several years while there were a few substantial changes. I've now [added a Changes section](https://github.com/w3c/csswg-drafts/commit/90af273474d5aa62f5c43705e13597b44abf8ada) and [annotated the spec with WPTs](https://github.com/w3c/csswg-drafts/pull/13104).

So I request to publish a new WD once the PR with the WPTs is merged.

Sebastian

**Comment by SebastianZ** [comment]
Like the previous specs, [CSS Logical 1](https://www.w3.org/TR/css-logical-1/) wasn't published since 2018 while some substantial changes were made to it. I've now [updated its Changes section](https://github.com/w3c/csswg-drafts/commit/fdf47222624f9eb8738d369763c021a320060115) and [added WPT coverage](https://github.com/w3c/csswg-drafts/commit/607e8b375e8cb91a09c9d01833e3c1dd654a750e).

Therefore, I request on @atanassov's and @fantasai's behalf to publish a new WD.

Sebastian

**Comment by SebastianZ** [comment]
And another spec that didn't get an updated WD for many years: [CSS Transitions 1](https://www.w3.org/TR/css-transitions-1/). Like for the specs mentioned earlier, I've [added a proper Changes section](https://github.com/w3c/csswg-drafts/commit/4f85c6fb72aef9eacac3e3bd523c85c266cfb949) and [annotated it with WPTs](https://github.com/w3c/csswg-drafts/pull/13133).

The spec. didn't receive that many substantial changes over the years, though given the long timespan to the last publication, I think it's time to publish a new WD.
So, I request to publish a new WD once the WPTs are merged if there's no objection (especially from its editors @dbaron, @grorg, and @birtles).

Sebastian

**Comment by SebastianZ** [comment]
CSS Borders 4 received a significant amount of [changes](https://drafts.csswg.org/css-borders-4/#changes-20250722) since its [FPWD in July this year](https://www.w3.org/TR/css-borders-4/). So we should probably also publish a new WD of it.
@fantasai @LeaVerou @noamr

Sebastian

**Comment by SebastianZ** [comment]
I've now added the latest requests to the initial comment, so it's easier to follow what's requested.

Sebastian

**Comment by SebastianZ** [comment]
[CSS Shadow Parts 1](https://www.w3.org/TR/css-shadow-parts-1/) is another spec that didn't get a new WD for a long time. Like for the other specs, I've now added a [Changes section](https://github.com/w3c/csswg-drafts/commit/856bb2f18b640872088d8f5cfbdbe4588852f116) and [WPT coverage](https://github.com/w3c/csswg-drafts/commit/93da5257077744e994f4a7b98a186f1d3f943f45).

There were a few substantial changes since its last publication, so requesting to publish a new WD.
@tabatkins @fergald 

Sebastian

**Comment by SebastianZ** [comment]
I've now updated the Changes sections of [CSS UI 3](https://drafts.csswg.org/css-ui-3/) and [CSS UI 4](https://drafts.csswg.org/css-ui-4/) and updated the WPT coverage, plus added a Changes from L3 section to L4. See https://github.com/w3c/csswg-drafts/commit/49e7e33709aa2ce283a3c7c898cf90b00778e636, https://github.com/w3c/csswg-drafts/commit/eb59c637958210ea01a6200b481371092651882c, https://github.com/w3c/csswg-drafts/commit/a1149d802097947e6b647946db92c394a362f0e3, and https://github.com/w3c/csswg-drafts/commit/9b128ba6b87f4660d4edf1696acfd9a250b1adb1.

[CSS UI 3](https://www.w3.org/TR/css-ui-3/) is already a recommendation since 21 June 2018, though there were a few changes made to it since then. So I think it is worth publishing an amended version of it.

And the [last published WD of CSS UI 4](https://www.w3.org/TR/css-ui-4/) from 16 March 2021 is totally outdated and missing key features like the `pointer-events` or `user-select` properties. So I believe it urgently needs an update.

Feedback on that is very welcome, especially from their editors @tantek, @frivoal, and @tabatkins.

Sebastian

**Comment by svgeesus** [comment]
> [CSS UI 3](https://www.w3.org/TR/css-ui-3/) is already a recommendation since 21 June 2018, though there were a few changes made to it since then. So I think it is worth publishing an amended version of it.

It wasn't published as an updatable Rec however (that wasn't a thing back then) so it would need to go CR Snapshot and then Rec. Are the changes sufficient to warrant that, or should we be focusing on getting CSS UI 4 updated and moved forward?

@frivoal (to check my first point and give advice on the second).

**Comment by svgeesus** [comment]
Content-3 requested 1 Dec, expected 4 Dec.

**Comment by SebastianZ** [comment]
And @svgeesus obviously also requested the CSS Logical 1 publication 1 Dec, expected 4 Dec. Thanks!

Sebastian

**Comment by SebastianZ** [comment]
@tabatkins @svgeesus In https://github.com/w3c/csswg-drafts/issues/12704#issuecomment-3249801090 we resolved on republishing CSS Nesting 1. Could you please request a publication? (That also [allows us to add it to CSS 2025](https://github.com/w3c/csswg-drafts/issues/9793#issuecomment-2518138682).)

Sebastian

**Comment by svgeesus** [comment]
- [Content 3 published 4 Dec](https://www.w3.org/TR/2025/WD-css-content-3-20251204/)
- [Logical 1 published 4 Dec](https://www.w3.org/TR/2025/WD-css-logical-1-20251204/)

**Comment by svgeesus** [comment]
Tables 3 requested 12 Dec, expected 16 Dec
Borders 4 requested 12 Dec, expected 16 Dec
Shadow Parts 1 requested 12 Dec, expected 16 Dec

**Comment by SebastianZ** [comment]
[Tables 3](https://www.w3.org/TR/css-tables-3/), [Borders 4](https://www.w3.org/TR/css-borders-4/) and [Shadow Parts 1](https://www.w3.org/TR/css-shadow-parts-1/) were published today. Thanks a lot @svgeesus!

That leaves us with Transitions 1 and UI 4.

Sebastian

**Comment by svgeesus** [comment]
> That leaves us with Transitions 1 and UI 4.

I will try to get those done before the publishing moratorium.

**Comment by SebastianZ** [comment]
And I have another spec that should get an updated WD: [CSS Rhythmic Sizing 1](https://drafts.csswg.org/css-rhythm-1/). It had a [FPWD back in 2017](https://www.w3.org/TR/css-rhythm-1/).
There weren't _that_ many changes since then, though at least a few significant enough to warrant a new WD. The [Changes section](https://drafts.csswg.org/css-rhythm-1/#changes) was mostly up-to-date, from what I could see. I've just added [two smaller points](https://github.com/w3c/csswg-drafts/commit/6da935cfd9bc44822c3656e05f310d382459e92a) to it plus added the [WPT coverage](https://github.com/w3c/csswg-drafts/commit/35ca36d8e01f11738010a5d441eeeb27e22e5a1e).

Sebastian

**Comment by svgeesus** [comment]
Transitions 1 requested 7 Jan, expected 8 Jan.

**Comment by svgeesus** [comment]
[Transitions 1 published 8 Jan](https://www.w3.org/TR/2026/WD-css-transitions-1-20260108/)

**Comment by SebastianZ** [comment]
In https://github.com/w3c/csswg-drafts/issues/12704#issuecomment-3249801090 we resolved to publish a new WD of CSS Nesting 1, which didn't happen yet.

(I verified that the list of changes and the WPTs are up-to-date and that there are no Bikeshed issues.)

Sebastian

**Comment by SebastianZ** [comment]
We also (asyncronously) [resolved to publish another WD of Selectors 4](https://github.com/w3c/csswg-drafts/issues/12799#issuecomment-3628969842) in preparation for moving it to CR. So added it to the list.

As several features of level 4 got deferred to level 5 for this, I also request to publish an FPWD of Selectors 5.
I will try to bring the WPT coverage up-to-date before the publication.

Sebastian

**Comment by SebastianZ** [comment]
[UI 4 published 20 Jan](https://www.w3.org/TR/2026/WD-css-ui-4-20260120/)

Thanks, @svgeesus!

Sebastian

**Comment by svgeesus** [comment]
- [Selectors 4 republished 22 Jan 2026](https://www.w3.org/TR/2026/WD-selectors-4-20260122/)
- [CSS Nesting 1 republished 22 Jan 2026](https://www.w3.org/TR/2026/WD-css-nesting-1-20260122/)

**Comment by svgeesus** [comment]
Removing link to the (Last Updated 2021-12-24) CSSWG Spec Publications Status because it is 5 years out of date.

**Comment by SebastianZ** [comment]
I've just updated the Changes section of [Media Queries 5](https://drafts.csswg.org/mediaqueries-5/). And since there have been a bunch of significant updates since its [last publication back in 2021](https://www.w3.org/TR/mediaqueries-5/), I believe it's time to publish a new WD.

Media Queries 4 would also benefit possibly also needs a new publication, though I didn't check, yet, whether its changes section needs any updates.

Sebastian

**Comment by SebastianZ** [comment]
> Media Queries 4 would also benefit possibly also needs a new publication, though I didn't check, yet, whether its changes section needs any updates.

Ok, I've verified that the changes section is up-to-date. Besides a bunch of editorial changes, there are only two substantial ones, which were already listed. The WPT annotations are also up-to-date. Therefore, I think we should publish a new CRD.

Sebastian

**Comment by svgeesus** [comment]
@astearns so the current list of things needing resolutions to publish, (and are ready for publication)

- CRD of [Media Queries 4](https://drafts.csswg.org/mediaqueries-4/)
- WD of [Media Queries 5](https://drafts.csswg.org/mediaqueries-5/)
- WD of [CSS Rhythm 1](https://drafts.csswg.org/css-rhythm-1/)
- FPWD of [Selectors 5](https://drafts.csswg.org/selectors-5/)

**Comment by svgeesus** [comment]
FPWD request will not be discussed until after the CSSWG call; if we do not get consensus to publish on Wednesday then I can pause it.

**Comment by SebastianZ** [comment]
I have two more specs that I think should get republished: [CSS Sizing 3](https://drafts.csswg.org/css-sizing-3/) and [CSS Sizing 4](https://drafts.csswg.org/css-sizing-4/).

I've updated their changes sections and added WPT annotations. Level 3 mainly got one addition with `calc-size()` plus some clarifications, while level 4 had some major updates.

Sebastian

**Comment by svgeesus** [comment]
- [Selectors Level 5 FPWD, published 17 Feb](https://www.w3.org/TR/2026/WD-selectors-5-20260217/)
- [CSS Rhythmic Sizing Module Level 1, published 17 Feb](https://www.w3.org/TR/2026/WD-css-rhythm-1-20260217/)

**Comment by svgeesus** [comment]
- [Media Queries 4 CRD published 19 Feb](https://www.w3.org/TR/2026/CRD-mediaqueries-4-20260219/)
- [Media Queries 5 WD published 19 Feb](https://www.w3.org/TR/2026/WD-mediaqueries-5-20260219/)

**Comment by kbabbitt** [comment]
Agenda+ to republish WD of [Gap Decorations 1](https://drafts.csswg.org/css-gaps-1/). All of the major changes we've talked about over the last few months have been incorporated, and I just finished up an editorial cleanup pass.

**Comment by svgeesus** [comment]
LGTM

**Comment by kbabbitt** [comment]
[Gaps 1 republished 27 Feb](https://www.w3.org/TR/2026/WD-css-gaps-1-20260227/). I have also moved the [explainer](https://github.com/w3c/csswg-drafts/blob/main/css-gaps-1/explainer.md) into the csswg-drafts repo.

**Comment by frivoal** [comment]
I'd like to propose republishing CSS-UI-3 as an updated REC.

Compared to the previous publication, it contains three normative changes marked as "proposed corrections". I've taken the liberty of marking them as _proposed_ rather than _candidate_, though we could argue about that. In any case, we'll need to document that they're adequately tested, and solicit appropriately wide review, before we can fold them in normatively. We may chose to publish them as candidate corrections first until that's done, and then republish as _proposed_ and run the corresponding [Last Call for Review of Proposed Amendments](https://www.w3.org/policies/process/#last-call-review). The Process does not require it, but AC Reps might push back and ask to see these steps first before approving.

I consider all other changes since last publication as REC to be editorial, though some were quite extensive. In particular, I'd like to call attention to https://github.com/w3c/csswg-drafts/commit/1a57a4025c8778dab45ef8d5fbd482e4d6c4ebce. It did introduce [a bug](https://github.com/w3c/csswg-drafts/issues/13001), which is now [fixed](https://github.com/w3c/csswg-drafts/pull/13018), as well as some redundant text, which is now [fixed](https://github.com/w3c/csswg-drafts/commit/d53db3c9b220d30d7ac91b82029cf6bbd49c2c00) too. With these two fixes in place, I do think it is editorial, but I'd encourage others to double check.

The [change section](https://drafts.csswg.org/css-ui-3/#changes-2018-06-21) is up to date.

**Comment by kbabbitt** [comment]
I'm wondering if we should additionally backport the css-ui-4 change in [commit 09f11f2](https://github.com/w3c/csswg-drafts/commit/09f11f2ef9964e29fb1308bfe48a5168a82f14b9#diff-cc2f666bf7d93dcefcc3fffc4761793a54fba00c03b53b9dbdf81f5056f0b56e) to css-ui-3. It's a (minor) breaking change between the two levels that three major engines are shipping [[1](https://chromestatus.com/feature/5133099851317248)] [[2](https://bugzilla.mozilla.org/show_bug.cgi?id=1998285)] [[3](https://bugs.webkit.org/show_bug.cgi?id=304940)].

**Comment by svgeesus** [comment]
Especially with the [proposed backport](https://github.com/w3c/csswg-drafts/issues/6900#issuecomment-4040806271), I support republishing with Candidate Corrections sooner rather than later, and republishing again with Proposed Corrections once adequate testing has been documented.

**Comment by frivoal** [comment]
> I'm wondering if we should additionally backport the css-ui-4 change in [commit 09f11f2](https://github.com/w3c/csswg-drafts/commit/09f11f2ef9964e29fb1308bfe48a5168a82f14b9#diff-cc2f666bf7d93dcefcc3fffc4761793a54fba00c03b53b9dbdf81f5056f0b56e) to css-ui-3

Good point. The original resolution did not limit the fix to level 4, so I went ahead and did it without waiting for a further resolution. See https://github.com/w3c/csswg-drafts/commit/7ce3eb5fb45cb89ffff88d396390261f684eb49b

> I support republishing with Candidate Corrections sooner rather than later, and republishing again with Proposed Corrections once adequate testing has been documented.

Agreed, and updated accordingly. See https://github.com/w3c/csswg-drafts/commit/4f1d355c9a32cf96bf2e70cc33e3008ca93c6287

**Comment by frivoal** [comment]
Separately from the above, I would also like to propose moving https://drafts.csswg.org/css-image-animation-1/ to FPWD, possibly after addressing https://github.com/w3c/csswg-drafts/issues/13641, though that could also be done after publication. Since the group has adopted it as an ED, I've made the following three changes based on the discussions we had:
* https://github.com/w3c/csswg-drafts/pull/13636
* https://github.com/w3c/csswg-drafts/pull/13638
* https://github.com/w3c/csswg-drafts/pull/13640

I think this (especially the addition of the distinction between `stopped` and `paused`) fleshes out the scope enough, and that we should be good to publish. I'd kick off TAG and horizontal review once we do.

**Comment by svgeesus** [comment]
[CSS UI 3 published 7 April 2026](https://www.w3.org/TR/2026/REC-css-ui-3-20260407/)

**Comment by svgeesus** [comment]
[CSS Image Animation published 9 April 2026](https://www.w3.org/TR/2026/WD-css-image-animation-1-20260409/)

**Comment by svgeesus** [comment]
> <TabAtkins> kbabbitt: looking thru changelog, some autolinks aren't working. contain-intrinsic-size links are sometimes not working
> <TabAtkins> astearns: so proposed resolution, fix the Changes link and then repub WD of Sizing 3 and 4
> <TabAtkins> RESOLVED: fix the Changes link and then repub WD of Sizing 3 and 4

@tabatkins @fantasai @jensimmons I'm holding off on publication until I see checkins for the fixes to Changelog.

**Comment by kbabbitt** [comment]
Agenda+ to republish Gaps Level 1 as WD. [Draft](https://drafts.csswg.org/css-gaps-1/) | [Changes](https://drafts.csswg.org/css-gaps-1/#changes=20260227)

**Comment by kbabbitt** [comment]
[Gaps 1 WD republished 24 June 2026](https://www.w3.org/TR/2026/WD-css-gaps-1-20260624/)

**Comment by svgeesus** [comment]
[CSS Linked Parameters FPWD published 14 July](https://www.w3.org/TR/2026/WD-css-link-params-1-20260714/)
